### DISCUSSION TITLE
Can't use plugins: "Unrecognized request argument supplied: functions"

### DISCUSSION THREAD

**Discussion by jacobstein123** [danny-avila/LibreChat#1614] [Troubleshooting] [answered]
When I try to use the DALL-E or DALL-E 3 plugin, every request results in the same error:

> Something went wrong. Here's the specific error message we encountered: Encountered an error while attempting to respond: 404 Unrecognized request argument supplied: functions

I've tried a number of different models, and I always get the same error. I've confirmed I can use my OpenAI API Key for DALL-E using the Python API.

I've searched through the issues on GitHub and haven't found anyone else experiencing this, so I'm wondering if I'm doing something wrong or if this is a new issue (I updated today, but never tried using plugins before updating).

Any help would be appreciated!

**Comment by danny-avila** [comment, marked as answer]
Thanks for checking out the project. Both are working for me so maybe there's something up with the exact model you have selected or the service you are accessing DALL-E through

Other than the selected models, Do you have any of these fields filled out? They are optional, since you can provide the key from the frontend.

```bash
# DALL·E
#----------------
# DALLE_API_KEY= # Key for both DALL-E-2 and DALL-E-3
# DALLE3_API_KEY= # Key for DALL-E-3 only
# DALLE2_API_KEY= # Key for DALL-E-2 only
# DALLE3_SYSTEM_PROMPT="Your DALL-E-3 System Prompt here"
# DALLE2_SYSTEM_PROMPT="Your DALL-E-2 System Prompt here"
# DALLE_REVERSE_PROXY= # Reverse proxy for DALL-E-2 and DALL-E-3
# DALLE3_BASEURL= # Base URL for DALL-E-3
# DALLE2_BASEURL= # Base URL for DALL-E-2
```

Check the completion model

![image](https://github.com/danny-avila/LibreChat/assets/110412045/129ba787-1a8c-466d-8ead-2f40117a97f1)

and the agent model

![image](https://github.com/danny-avila/LibreChat/assets/110412045/f210ee17-3ef5-418d-8fa0-38e68dcd544b)

Note you should have those 2 switches on, for functions and skip completion, like I do

**Reply by jacobstein123** [reply]
I didn't change anything, but it randomly started working. Must have been a brief intermittent issue on OpenAI's end that they fixed. I'll mark this as complete.

**Reply by danny-avila** [reply]
great, glad it started working!

**Reply by artificialartofficial** [reply]
I had this same issue and realized my agent was set to one of the older models that no longer works with DALL-E. Thanks for posting this was helpful for debugging!
